optuna/visualization/_pareto_front.py

```diff
@@ -74,196 +91,142 @@ def objective(trial):
 
     _imports.check()
 
-    if len(study.directions) == 2:
-        return _get_pareto_front_2d(study, target_names, include_dominated_trials, axis_order)
-    elif len(study.directions) == 3:
-        return _get_pareto_front_3d(study, target_names, include_dominated_trials, axis_order)
-    else:
+    n_dim = len(study.directions)
+    if n_dim not in (2, 3):
         raise ValueError("`plot_pareto_front` function only supports 2 or 3 objective studies.")
 
-
-def _get_non_pareto_front_trials(
-    study: Study, pareto_trials: List[FrozenTrial]
-) -> List[FrozenTrial]:
-
-    non_pareto_trials = []
-    for trial in study.get_trials():
-        if trial.state == TrialState.COMPLETE and trial not in pareto_trials:
-            non_pareto_trials.append(trial)
-    return non_pareto_trials
-
-
-def _get_pareto_front_2d(
-    study: Study,
-    target_names: Optional[List[str]],
-    include_dominated_trials: bool = False,
-    axis_order: Optional[List[int]] = None,
-) -> "go.Figure":
     if target_names is None:
-        target_names = ["Objective 0", "Objective 1"]
-    elif len(target_names) != 2:
-        raise ValueError("The length of `target_names` is supposed to be 2.")
-
-    trials = study.best_trials
-    n_best_trials = len(trials)
-    if len(trials) == 0:
-        _logger.warning("Your study does not have any completed trials.")
+        target_names = [f"Objective {i}" for i in range(n_dim)]
+    elif len(target_names) != n_dim:
+        raise ValueError(f"The length of `target_names` is supposed to be {n_dim}.")
+
+    if constraints_func is not None:
+        feasible_trials = []
+        infeasible_trials = []
+        for trial in study.get_trials(states=(TrialState.COMPLETE,)):
+            if all(map(lambda x: x <= 0.0, constraints_func(trial))):
```

Comment:
I'm a bit conservative to add a new method to `FrozenTrial` since`Trial` does not have the corresponding method. And also the type signature will have a nested structure i.e., the method of `FrozenTrial` instance takes a function of `FrozenTrial` instance.

Reply:
I guess it can be discussed in a different ticket, but just to elaborate my initial thought, was to create a private free function.